modules/fbs/fbs.module

```diff
@@ -495,6 +495,54 @@ function fbs_form_profile2_form_alter(&$form, &$form_state) {
     '#markup' => $fee_sms,
     '#weight' => 3,
   );
+  
+  // Add validation to email.
+  $form['profile_provider_fbs']['field_fbs_mail']['#element_validate'] = array('fbs_form_profile2_mail_validate');
+  // Add validation to phone.
+  $form['profile_provider_fbs']['field_fbs_phone']['#element_validate'] = array('fbs_form_profile2_phone_validate');
+  // Add submit handler, to redirect to same URL.
+  $form['#submit'][] = 'fbs_form_profile2_rebuild_submit';
+}
+
+/**
+ * Mail validation for fbs_form_profile2
+ */
+function fbs_form_profile2_mail_validate($element, $form_state) {
+  $email = $form_state['input']['profile_provider_fbs']['field_fbs_mail']['und'][0]['value'];
+  if (!empty($email)) {
+    if (!filter_var($email, FILTER_VALIDATE_EMAIL) && !preg_match('/@.+\./', $email)) {
```

Comment:
Why not just use "!filter_var($email, FILTER_VALIDATE_EMAIL)"? 
Adding the "!preg_match('/@.+\./', $email)" expression will allow email adresses like 'joe@example.com ' or 'foo @ bar . bas' to pass.
Btw, it might be a good idea to strip whitespace from start and end - it's a common copy/paste error.